pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/PartitionSegmentPruner.java

```diff
@@ -114,20 +120,25 @@ private PartitionInfo extractPartitionInfoFromSegmentZKMetadataZNRecord(String s
     } catch (Exception e) {
       LOGGER.warn("Caught exception while extracting segment partition metadata for segment: {}, table: {}", segment,
           _tableNameWithType, e);
-      return INVALID_PARTITION_INFO;
+      return INVALID_COLUMN_PARTITION_INFO;
     }
 
-    ColumnPartitionMetadata columnPartitionMetadata =
-        segmentPartitionMetadata.getColumnPartitionMap().get(_partitionColumn);
-    if (columnPartitionMetadata == null) {
-      LOGGER.warn("Failed to find column partition metadata for column: {}, segment: {}, table: {}", _partitionColumn,
-          segment, _tableNameWithType);
-      return INVALID_PARTITION_INFO;
+    Map<String, PartitionInfo> columnPartitionInfo = new HashMap<>();
```

Comment:
Suggest returning a singleton map to accelerate the map lookup when there is only one entry in the map